kat/kat/harness.py

```diff
@@ -26,6 +26,26 @@
 from .parser import dump, load, Tag
 
 
+class TestImage:
+    def __init__(self, *args, **kwargs) -> None:
+        self.images: Dict[str, str] = {}
+
+        default_registry = os.environ.get('TEST_SVC_REGISTRY', 'quay.io/datawire/test_services')
+        default_version = os.environ.get('TEST_SVC_VERSION', '0.0.2')
+
+        for svc in ['auth', 'auth-tls', 'ratelimit', 'shadow', 'stats']:
+            key = svc.replace('-', '_').upper()
+
+            image = os.environ.get(f'TEST_SVC_{key}', f'{default_registry}:test-{svc}-{default_version}')
+
+            self.images[svc] = image
+
+    def __getitem__(self, key: str) -> str:
+        return self.images[key]
+
+
+GLOBAL_TEST_IMAGE = TestImage()
```

Comment:
Is this better than just returning the `images` dict?  Having a class that implements `__getitem__` as just wrapping a dict feels funny to me (but I'm not a pythonista).

Reply:
hmm, I wish I could tel GitHub to include the entire `TestImage` class as context for that comment :/

Reply:
Hmmmm... so basically what's going on here is that, for whatever reason, I'm more comfortable having a singleton class instance as a global than I am with inlining the code in `TestImage.__init__` at global scope and having it create a global dict. But I didn't want to create a new `TestImage` for every `Node` because there can be a lot of `Node`s...

It's not exactly a slam-dunk either way. 😐